lib/internal/test_runner/test.js

```diff
@@ -605,6 +605,14 @@ class Test extends AsyncResource {
           throw new ERR_INVALID_ARG_TYPE('options.concurrency', ['boolean', 'number'], concurrency);
     }
 
+    // Take timeout from cli
+    const cliTimeout = this.config.timeout;
```

Comment:
This shouldn't happen unconditionally. This should only happen when the `timeout` option to the test is not provided.

Reply:
`this.timeout` will be first set to parent's timeout above - and `this.timeout` will also be overwritten below when `timeout` from options is provided. I guess the readability of this line is poor and confusing.

Reply:
I'm saying that this should only happen if `timeout == null`.

Reply:
Thanks for the suggestion, absolutely!

Reply:
I made it into an `else if` could you take another look please? 🙏